### DISCUSSION TITLE
Client Updates

### DISCUSSION THREAD

**Discussion by HeyVern** [Ylianst/MeshCentral#5499] [Q&A]
Everytime a new server release comes out, I update immediately (like Christmas each time).  But I noticed that my client dates don't seem to be updating...so my question is.... If server updates prompt client updates do those occur automatically...and then if so, what dates/versions should be I be seeing if I am reviewing client info to confirm they are on the latest client version (Winx64).  Thanks.

**Comment by si458** [comment]
The client updates are separate to the server updates
The meshcentral version can just be new features, bug fixes, code changes 1.1.15 for example
Where as the agent version only really changes when something major needs fixing in the agent exes
The agent updates will get pushed out automatically is you have this enabled (by default it is)

I do think we need a panel tho to show the latest agent version that's available, rather than just showing what version is installed on the remote device, as u don't no if the is a new agent or not (Can u open a enhancement for this please?)

**Comment by HeyVern** [comment]
Thank you.  Will do.  You distilled down the issue I was asking about well.  Not knowing the latest client version number is what led me to ask.  This enhancement would solve the issue.

**Reply by si458** [reply]
Would you want a message on the device details page saying 'you need to update the agent to xyz'
Or
Would you want an option in my server page, to just show you what each version is for each os, then u know for reference?

**Reply by HeyVern** [reply]
It would make since for both, but if only one location 'My Server Page'

**Comment by NiceGuyIT** [comment]
I read somewhere that the agent version is tied to the server version. If the server is updated _and an agent update is included in the release_, then the agents are updated automatically.

- #4671

The exception is if the `noagentupdate=1` flag is set which will not update the agent when the server updates.

- #2994

I realize this conversation is about verifying the agent version is in sync with the server version.

**Reply by si458** [reply]
You are indeed correct, if the agent gets updated on a meshcentral release, a json file verifies the hash matches the remote hash of the exe, if different then auto update if enabled!

**Reply by si458** [reply]
https://github.com/Ylianst/MeshCentral/blob/master/agents/hashagents.json

**Reply by NiceGuyIT** [reply]
Found the issue I was looking for.

- #1979

> The server will automatically update agents to match the agents in the server version, so if you downgrade a server, all agents will also be downgraded if needed. Starting with v0.6.85 the server has a new set of MeshAgents and you can use the "versions" command in the agent console to see the agent compile and commit time . Hope that helps.

**Comment by mwllgr** [comment]
Cross-referencing this here, as it looks like the agent is not really receiving any updates right now: https://github.com/Ylianst/MeshAgent/issues/214
